internal/testrunner/runners/policy/policy.go

```diff
@@ -0,0 +1,226 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package policy
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pmezard/go-difflib/difflib"
+	"gopkg.in/yaml.v3"
+
+	"github.com/elastic/elastic-package/internal/common"
+	"github.com/elastic/elastic-package/internal/testrunner"
+)
+
+func dumpExpectedAgentPolicy(ctx context.Context, options testrunner.TestOptions, testPath string, policyID string, expectedRevision int) error {
+	policy, err := options.KibanaClient.DownloadPolicy(ctx, policyID)
+	if err != nil {
+		return fmt.Errorf("failed to download policy %q: %w", policyID, err)
+	}
+
+	d, err := cleanPolicy(policy, policyEntryFilters)
+	if err != nil {
+		return fmt.Errorf("failed to prepare policy to store")
```

Comment:
Nit. Show the error in this message
```suggestion
		return fmt.Errorf("failed to prepare policy to store: %w", err)
```